compiler/optimizer/VPHandlers.cpp

```diff
@@ -12216,11 +12214,14 @@ TR::Node *constrainArrayStoreChk(OMR::ValuePropagation *vp, TR::Node *node)
          if (base->getFirstChild()->getOpCode().isArrayRef())
             base = base->getFirstChild()->getFirstChild();
          }
+
+         valueIsFromSameArray = (base == arrayRef);
```

Comment:
Since this is value propagation I wonder if this test could just have been on the value numbers being the same, rather than the nodes being the same.

Reply:
I see that change still doesn't help with my example in Local Value Propagation, as there is no value numbering available.  However, it would still help if that case was encountered by Global Value Propagation.